kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/PartitionUtils.java

```diff
@@ -248,6 +251,44 @@ private static Expression rewritePartitionColumnRef(
         return expression;
     }
 
+    /**
+     * Get the target directory for writing data for given partition values.
+     *
+     * @param dataRoot          Root directory where the data is stored.
+     * @param partitionColNames Partition column names. We need this to create the target directory
+     *                          structure that is consistent levels of directories.
+     * @param partitionValues   Partition values to create the target directory.
+     * @return Target directory path.
+     */
+    public static String getTargetDirectory(
+            String dataRoot,
+            List<String> partitionColNames,
+            Map<String, Literal> partitionValues) {
+        Path targetDirectory = new Path(dataRoot);
+        for (String partitionColName : partitionColNames) {
+            Literal partitionValue = partitionValues.get(partitionColName);
+            checkArgument(
+                    partitionValue != null,
```

Comment:
can any NULL values be partition values? i.e.
`part1=helloworld`
`part1=NULL`

Reply:
yes, then can be. In `AddFile.partitionValues` they are not written (empty is considered as `null` value`)